lib/ansible/parsing/mod_args.py

```diff
@@ -232,10 +249,10 @@ def _normalize_new_style_args(self, thing, action):
             # this can happen with modules which take no params, like ping:
             args = None
         else:
-            raise AnsibleParserError("unexpected parameter type in action: %s" % type(thing), obj=self._task_ds)
+            raise AnsibleParserError(f"unexpected parameter type in action: {type(thing)}", obj=self._task_ds)
         return args
 
-    def _normalize_old_style_args(self, thing):
+    def _normalize_old_style_args(self, thing: dict[str, t.Any] | str | bytes | None) -> tuple[str, dict[str, t.Any]]:
```

Comment:
You can also reuse that type alias here. And there's the same problem with bytes:
```suggestion
    def _normalize_old_style_args(self, thing: PreParsedArgs | str | None) -> tuple[str, PreParsedArgs]:
```